--- a/package/boot/uboot-mediatek/Makefile
+++ b/package/boot/uboot-mediatek/Makefile
@@ -469,6 +469,18 @@ define U-Boot/mt7981_jcg_q30-pro
   DEPENDS:=+trusted-firmware-a-mt7981-spim-nand-ddr3
 endef
 
+define U-Boot/mt7981_ikuai_q3000-ubi
+  NAME:=iKuai Q3000 (UBI)
+  BUILD_SUBTARGET:=filogic
+  BUILD_DEVICES:=ikuai_q3000-ubi
+  UBOOT_CONFIG:=mt7981_ikuai_q3000-ubi
+  UBOOT_IMAGE:=u-boot.fip
+  BL2_BOOTDEV:=spim-nand-ubi
+  BL2_SOC:=mt7981
+  BL2_DDRTYPE:=ddr3-1866
+  DEPENDS:=+trusted-firmware-a-mt7981-spim-nand-ubi-ddr3-1866
+endef
+
 define U-Boot/mt7981_konka_komi-a31
   NAME:=Konka KOMI A31
   BUILD_SUBTARGET:=filogic
@@ -1247,6 +1259,7 @@ UBOOT_TARGETS := \
 	mt7981_h3c_magic-nx30-pro \
 	mt7981_imou_hx21 \
 	mt7981_jcg_q30-pro \
+ 	mt7981_ikuai_q3000-ubi \
 	mt7981_konka_komi-a31 \
 	mt7981_netis_eap930-v1 \
 	mt7981_netis_nx30v2 \

--- a/package/boot/uboot-tools/uboot-envtools/files/mediatek_filogic
+++ b/package/boot/uboot-tools/uboot-envtools/files/mediatek_filogic
@@ -37,6 +37,7 @@ cudy,wr3000p-v1-ubi|\
 h3c,magic-nx30-pro|\
 imou,hx21|\
 jcg,q30-pro|\
+ikuai,q3000-ubi|\
 konka,komi-a31|\
 mercusys,mr90x-v1-ubi|\
 netcore,n60|\
